web-client/public/gene/api.js

```diff
@@ -169,34 +169,20 @@ let getEnsemblInfo = function (geneSymbol) {
 
 let getJasparInfo = function (geneSymbol) {
     return $.get({
-        url: serviceRoot + "/jaspar/api/v1/matrix/?tax_id=4932&format=json&search=" + geneSymbol,
+        url: serviceRoot + "/jaspar/api/v1/matrix/?tax_id=4932&format=json&name=" + geneSymbol.toUpperCase(),
         dataType: "json",
         beforeSend: function (xhr) {
             xhr.setRequestHeader("content-type", "application/json");
         },
     }).then(function (data) {
-        let value = undefined;
-        const nameCheck = (list) => {
-            if (list !== []) {
-                list.forEach((gene) => {
-                    if (gene.name === geneSymbol) {
-                        value = gene;
-                    }
-                });
-            }
-            return value;
-        };
-
-        return ((nameCheck(data.results) === undefined) ? {} :
+        return (data.results.length === 0) ? {} :
```

Comment:
Does the API guarantee that the response will have a `results` property, and that it will be an array? Make sure that those conditions are certain. Otherwise, you will need additional guard code to check whether `data.results` is undefined/null before inquiring about its length.

Reply:
@johnllopez616 will fix this and update the pull request and send it to @dondi for one more review.  @johnllopez616 and @mihirsamdarshi will get together to do the pull request checklist and walk through updating the server.